### ISSUE TITLE
I18N Fix GData importer for better translatability

### ISSUE THREAD

**Issue by tfmorris** [OpenRefine/OpenRefine#3312]
This is an issue that was raised by translators on Weblate.

Currently the gdata-import module has sentences split up into multiple fragments, such as:

"Please", "sign in and and authorize", "to access your Google data." with the middle phrase being a button label. This mandates an ordering which doesn't work in all languages.

A better structure would be: "Please sign-in and authorize to access your Google data:" followed by a button "Sign In"

A similar problem exists for "Re-sign in", "with another account" where the first fragment is a button label.

The key `gdata-import/sign-in` can be used to find the appropriate strings and code.

A related issue is the structure of

     "Ignore first" ____ "line(s) at beginning of file"

but that is shared with many importers, so probably should be addressed separately.

**Comment by harshitmalik03** [comment]
Hey, Can i work on this ??? I might need a bit of help as im new to this.

**Comment by wetneb** [comment]
Thanks for volunteering! Let us know where you need help.

**Comment by harshitmalik03** [comment]
I need someone to assist me on how do i get going in such a contribution. Ive read all the details and i can do the work but as its my first task i need somebody to guide me on call (if possible).

**Comment by wetneb** [comment]
I am not available to get on a call at the moment, but I hope you can find the information you need here:
https://github.com/OpenRefine/OpenRefine/wiki/Your-first-pull-request
If there is anything missing there, I would be interested to know what, so I can improve this documentation directly.

**Comment by harshitmalik03** [comment]
Actually i do know how to create a pull request. Im just having trouble
solving issues because im new to this.... I can talk to you whenever you
are free and available but it would be great if someone can help me out.

**Comment by wetneb** [comment]
If you are able to ask a precise question here, I would be happy to try to answer it.

**Comment by harshitmalik03** [comment]
Okay Ill try! Im a student from India and I was aiming for GSoC 21 and
hence wanted to try some open source projects. Its been a month now and ive
learnt git and stuff but im not getting on what i actually need to do.
Sometimes the task is to update some examples or something like that but
i'm not able to execute these..... Is there anything that i need to learn
before I start doing this? I can code in java and python and can do basic
documentation stuff too. It would be god if you could help me out.
Thanks !

**Comment by wetneb** [comment]
I would recommend that you follow the steps described here:
https://github.com/OpenRefine/OpenRefine/wiki/Your-first-pull-request
If you get stuck at one of these steps, tell me which one, what the concrete problem is, which bit of information you are missing and where you looked for it.

**Comment by harshitmalik03** [comment]
Currently the gdata-import module has sentences split up into multiple fragments, such as:

"Please", "sign in and and authorize", "to access your Google data." with the middle phrase being a button label. This mandates an ordering which doesn't work in all languages.

Now where do i find this issue #3312 ??? Im not able to get this in the whole code to change it.

**Comment by wetneb** [comment]
I am not sure what you mean. Issue #3312 is this issue you are commenting on, you have already found it.
If you are trying to find which part of the code you should change to fix it, you can follow this method (which is already mentioned in the guide linked above):
https://github.com/OpenRefine/OpenRefine/issues/3137#issuecomment-691649962

**Comment by S-Harshit** [comment]
@wetneb  I have question regarding this issue - Are the strings "Please", "sign in and authorize", "to access your Google data."  part of the login box that is supposed to show up after clicking on google data cause I am experiencing Issue #3054 after clicking on Google Data

**Comment by wetneb** [comment]
The login box is provided by Google itself (as part of the OAuth process) so these strings are not used there.
To enable Google login on your development version, you need to follow these steps:
https://github.com/OpenRefine/OpenRefine/wiki/Google-Extension

**Comment by singhakshita** [comment]
hey if nobody is working on this issue can i please work ?

**Comment by wetneb** [comment]
Sure, go ahead, thanks for volunteering!

**Comment by singhakshita** [comment]
@wetneb for some reason its never giving me this sign in option ,even after are removed all logged in account ,it is still showing me this .
<img width="1440" alt="Screenshot 2021-02-01 at 12 08 47 PM" src="https://user-images.githubusercontent.com/52077474/106423502-94d03380-6486-11eb-9144-d0f3492c2b7c.png">

can you tell me how to reproduce the issue if i m doing anything wrong

**Comment by wetneb** [comment]
Have you looked for errors in the OpenRefine server logs or in your browser's Javascript console?

**Comment by singhakshita** [comment]
yeah its not showing any error or any warning .
its just showing that spinner of loading 
that never loads
please guide me a bit

**Comment by singhakshita** [comment]
@wetneb  i tried to reproduce this issue but for some reason  i never get this option,it just shows me that spinner
so i m not able to fix this issue

**Comment by wetneb** [comment]
@singhakshita have you looked for errors in the OpenRefine server logs or in your browser's Javascript console?

**Comment by singhakshita** [comment]
yeah i saw it , initially it was showing some problems related to java version ,so i recloned the folder ,now it isn't showing any error ,but still its just showing spinner .
i have tried all possible ways to get rid of the  problem ,but now its not even showing any error

**Comment by wetneb** [comment]
Then I would try debugging this, either with your web browser's debugger or with the Java debugger via your IDE, depending on what the HTTP traffic between the frontend and the backend looks like.

**Comment by singhakshita** [comment]
<img width="1440" alt="Screenshot 2021-02-03 at 5 52 01 PM" src="https://user-images.githubusercontent.com/52077474/106746417-87fb3d80-6648-11eb-9bb8-19964a0de99b.png">

this is how my browser console looks like and also one more thing it also asks "there are some java files in the workspace do you want to import it " ,if i import it it starts showing 325 problems in my IDE console and no error in browser console.

if i dont import it shows no error any where

**Comment by singhakshita** [comment]
<img width="1440" alt="Screenshot 2021-02-03 at 6 12 15 PM" src="https://user-images.githubusercontent.com/52077474/106748574-70718400-664b-11eb-838f-d701a8bf0349.png">

these are the problems .that shows on only on importing these optional java files 
expect for google data rest everything works well

**Comment by harshitmalik03** [comment]
Hey
Im Harshit Malik , a freshman , wanting to learn open source contributions ... Im having trouble resolving issues on github... Could you please guide me upon this ? 
Thank you.

**Comment by S-Harshit** [comment]
> The login box is provided by Google itself (as part of the OAuth process) so these strings are not used there.
> To enable Google login on your development version, you need to follow these steps:
> https://github.com/OpenRefine/OpenRefine/wiki/Google-Extension

@singhakshita If you want to enable login box u should try the steps given in the above link .

**Comment by wetneb** [comment]
I am removing the "good first issue" tag from this issue because of the slight difficulty involved in setting up the Google extension for development. Given the struggles of various contributors above I don't think this is working out and we should not attract more inexperienced users to this issue.

**Comment by singhakshita** [comment]
@S-Harshit  thanks for the guidance ,i have set that up except for
"Add Java property definitions for your client id and secret key by adding -Dext.gdata.clientid=<yourid> and -Dext.gdata.clientsecret=... to your JVM arguments" 
this is not very clear ,can you please tell me if something has to imported or exported from this file and also how to write this file

**Comment by singhakshita** [comment]
@wetneb  i guess this problem is not because of the extension , i have set that up and put the credentials in the refine.ini
but still i am not getting sign in option also not getting any error 
please help me out with this .
also after reference to the issue  [https://github.com/OpenRefine/OpenRefine/issues/2383] i think [https://github.com/OpenRefine/OpenRefine/wiki/Google-Extension](url) needs some updation

**Comment by wetneb** [comment]
If you want to make changes to the wiki, you should be able to do so yourself directly:
https://github.com/OpenRefine/OpenRefine/wiki/Google-Extension/_edit

I do not know how to help you given the description of the problem you have. If I had the same problem myself I would just use the debugger to find out why you do not have the option to sign in. If you are not comfortable with using the debugger I would suggest you look at another issue.

**Comment by tfmorris** [comment]
This is blocked by #3054.
